### ISSUE TITLE
Unable to save model with optional fake fields

### ISSUE THREAD

**Issue by chancezeus** [Laravel-Backpack/CRUD#616]
I'm trying to save a model where (some of) the fake fields are optional, unfortunately this results in an sql exception (Column not found) for the fake field if it is not filled.

The reason (I suspect after looking through the code) for this is that the `isset($request[$fields[$k]['name']])` (starting from php 5.4) also returns false if the field inside equals `null`.

I think the compactFakeFields should be updated to always pull the field from the request array and to either allow null/empty values or at least move the `isset($request[$fields[$k]['name']])` inside the "if a field is defined as fake" check (because the way it is build now, a field basically get's _unfaked_ when the request variable is undefined/`null`).

**Comment by tabacitu** [comment]
Hi @chancezeus ,

Fake fields _should_ be allowed to be optional, I completely agree. But I just tested it and it worked fine for me. 

**Do you have your fake field mentioned in the ```$fillable``` property on your model?** If so, please remove it and leave only the actual db column (```extras``` or whatever), along with real fields. That should be the problem.

Hope it helps. Cheers!

**Comment by relincuish** [comment]
@tabacitu : I had the same issue above and adding the `extras` field to the model's `$fillable` property did the trick. In my situation, however, I was previously using the `$guarded` property; shouldn't they both work?

Before (failed w/ optional fake columns):
```php
protected $guarded = [
    'id',
    'created_at',
    'updated_at',
    'deleted_at'
];
// protected $fillable = [];
```

And after (works as expected):
```php
// protected $guarded = [];
protected $fillable = [
    'fillable',
    'columns',
    'including',
    'extras'
];
```

**Comment by promatik** [comment]
I was having this issue, just solved it and created a pull request #787.

The variable was arriving at `compactFakeFields` as null, so it won't be treated as a fake field, backpack would try to store it on its own column.
By replacing `isset` by `array_key_exists`, the key exists as null, so it's treated as a fake field.

**Comment by tabacitu** [comment]
Just merged @promatik 's PR into ```dev```. Will be available to everyone on master shortly. Thanks!
